storage/pruning/pruningStorer.go

```diff
@@ -538,7 +545,17 @@ func (ps *PruningStorer) changeEpoch(epoch uint32) error {
 		return err
 	}
 
-	err = ps.closeAndDestroyPersisters(epoch)
+	oldestEpochToKeep := computeOldestEpoch(metaBlock)
+	oldestEpochInCurrentSetting := ps.activePersisters[len(ps.activePersisters)-1].epoch
+	if oldestEpochToKeep > oldestEpochInCurrentSetting {
```

Comment:
add another constant - maxToKeepEvenIfOtherShardIsStucked -> to 5 for example.

Reply:
done